File: rust_vs_c++/classe/classe.cpp
// Hugo BERANGER - M2 MIAGE IA

#include <iostream>

using namespace std;

class Rectangle {
        int largeur, longueur;
    public:
        Rectangle (int x, int y) {
            largeur = x;
            longueur = y;
        }
        int aire () {
            return largeur*longueur;
        }
};

int main () {
    Rectangle rectangleObjet(3,4);
    cout << "c++  - rectangle aire: " << rectangleObjet.aire() << endl;
    return 0;
}

File: rust_vs_c++/classe/classe.rs
// Hugo BERANGER - M2 MIAGE IA

struct Rectangle {
    largeur: i32,
    longueur: i32,
}

impl Rectangle {
        fn aire(&self) -> i32 {
            self.largeur * self.longueur
        }
}

fn main() {
    let rectangle_objet = Rectangle {
        largeur: 3,
        longueur: 4,
    };
    println!("rust - rectangle aire: {}", rectangle_objet.aire());
}